### PR TITLE
Various tidying and replace text surface with glass

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
Addresses #159 #160

File: src/block.h [hunk 1]
```diff
@@ -7,10 +7,15 @@ enum block_type {
...
 enum surface_type {
-    surface_none,
-    surface_wall,
-    surface_grate,
-    surface_text,
+    surface_blocks_light = 0x40,
+    surface_blocks_air = 0x80,
+    surface_none = 0,
+
+    surface_wall = surface_blocks_light | surface_blocks_air,
```
Comment (mkfifo) [inline src/block.h]: will just need to add something more once we have more surfaces that can block air and/or light (using lower bits).

**Comment by chrisforbes** [comment]
Also addresses #158.

File: src/physics.h [hunk 2]
```diff
@@ -77,3 +77,15 @@ phys_raycast(glm::vec3 start, glm::vec3 end,
 generic_raycast_info
 phys_raycast_generic(glm::vec3 start, glm::vec3 end,
                      btCollisionObject *ignore, btCollisionWorld *world);
+
+static inline glm::vec3
```
Comment (mkfifo) [inline src/physics.h]: +1 to these helpers

File: src/ship_space.cc [hunk 3]
```diff
@@ -3,32 +3,6 @@
...
-{
-    unsigned int x = 0,
-                 y = 0,
-                 z = 0;
-
-    for( x = 0; x < xd; ++x ){
-        for( y = 0; y < yd; ++y ){
-            for( z = 0; z < zd; ++z ){
-                glm::ivec3 v(x, y, z);
-                this->chunks[v] = new chunk();
```
Comment (mkfifo) [inline src/ship_space.cc]: removing this and relying instead of ensure is a MUCH nicer interface.

**Comment by mkfifo** [comment]
Very happy with the code changes, unable to compile and test until I get home though.

**Pull request merged**